docs/external-node.md

```diff
@@ -0,0 +1,569 @@
+cannot directly access the `ClusterIP` address. Therefore, the `antrea` Service
+needs to become externally-accessible, by changing its type to `NodePort` or
+`LoadBalancer`.
+
+Since `antrea-agent` is running on an external Node which is not managed by
+Kubernetes, a configuration file needs to be present on each machine where the
+`antrea-agent` is running, and the path to this file will be provided to the
+`antrea-agent` as a command-line argument. Refer to the [sample configuration](../build/yamls/externalnode/conf/antrea-agent.conf)
+to learn the`antrea-agent` configuration options when running on an external Node.
+
+A further [section](#install-antrea-agent-on-vm) will provide detailed steps
+for running the `antrea-agent` on a VM.
+
+## The ExternalNode resource
+
+An example `ExternalNode` resource:
+
+```yaml
+apiVersion: crd.antrea.io/v1alpha1
+kind: ExternalNode
+metadata:
+  name: vm1
+  namespace: vm-ns
+  labels:
+    role: db
+spec:
+  interfaces:
+    - ips: [ "172.16.100.3" ]
+      name: ""
+```
+
+Note: **Only one interface is supported for Antrea v1.8**.
+
+### Name and Namespace
+
+The `name` field in an `ExternalNode` must be unique, which is the key for
+`antrea-controller` to identify an `antrea-agent`. The `ExternalNode` name is
+given to `antrea-agent` via an environment variable `NODE_NAME`, and
+`antrea-agent` will use the hostname to find the `ExternalNode` resource if it
+is not set.
+
+`ExternalNode` resource is `Namespace` scoped. The `Namespace` is provided to
+`antrea-agent` with option `externalNodeNamespace` in
+[antrea-agent.conf](../build/yamls/externalnode/conf/antrea-agent.conf).
+
+```yaml
+externalNodeNamespace: vm-ns
+```
+
+### Interfaces
+
+The `interfaces` field specifies the list of the network interfaces expected to
+be guarded by Antrea NetworkPolicy. At least one interface is required. Interface
+`name` or `ips` is used to identify the target interface. **The field `ips`
+must be provided in the CRD**, but `name` is optional. Multiple IPs on a single
+interface is supported. In the case that multiple `interfaces` are configured,
+`name` must be specified in every `interface`.
+
+`antrea-controller` creates an `ExternalEntity` for each interface whenever an
+an `ExternalNode` is created. The created `ExternalEntity` has the following
+characteristics:
+
+- It is configured within the same Namespace as the `ExternalNode`.
+- The `name` is generated according to the following principles:
+  - Use the `ExternalNode` name directly, if there is only one interface, and
+    interface name is not specified.
+  - Use the format `$ExternalNode.name-$hash($interface.name)[:5]` for other
+    cases.
+- The `externalNode` field is set with the `ExternalNode` name.
+- The `owner` is referring to the `ExternalNode` resource.
+- All labels added on `ExternalNode` are copied to the `ExternalEntity`.
+- Each IP address of the interface is added as an Endpoint in the `endpoints`
```

Comment:
Should we use "endpoint" instead of "Endpoint"?

Reply:
updated.